From ad736dcb4c30ea23da2363d13d385c0b34703124 Mon Sep 17 00:00:00 2001
From: halx99 <halx99@live.com>
Date: Sat, 25 May 2024 23:44:49 +0800
Subject: [PATCH] fix android armv7 build

---
 src/lib/ares_event_configchg.c | 16 +++++++++++++++-
 1 file changed, 15 insertions(+), 1 deletion(-)

diff --git a/src/lib/ares_event_configchg.c b/src/lib/ares_event_configchg.c
index dc6e465..eb53fe4 100644
--- a/src/lib/ares_event_configchg.c
+++ b/src/lib/ares_event_configchg.c
@@ -33,7 +33,21 @@ static void ares_event_configchg_reload(ares_event_thread_t *e)
   ares_reinit(e->channel);
 }
 
-#ifdef __linux__
+#ifdef __ANDROID__
+
+ares_status_t ares_event_configchg_init(ares_event_configchg_t **configchg,
+                                        ares_event_thread_t     *e)
+{
+  /* No ability */
+  return ARES_ENOTIMP;
+}
+
+void ares_event_configchg_destroy(ares_event_configchg_t *configchg)
+{
+  /* No-op */
+}
+
+#elif defined(__linux__)
 
 #  include <sys/inotify.h>
 
-- 
2.39.0.windows.2

